### ISSUE TITLE
OCR OOD metrics and PickScore reward collapse

### ISSUE THREAD

**Issue by Solitude-lyh** [Luo-Yihong/DGPO#10]
Hi authors, thanks for releasing the code. I'm having trouble reproducing the results from the paper and would appreciate your guidance.

I'm training with dgpo.py with num_image_per_prompt = 24 and num_groups = 24, with all other hyperparameters at the defaults.
On the OCR task, evaluating the step = 360 checkpoint, the in-domain OCR accuracy matches the paper, but the out-of-domain metrics on Drawbench are significantly lower than reported — which looks like reward hacking on the OCR signal, as shown in the screenshot.

<img width="236" height="74" alt="Image" src="https://github.com/user-attachments/assets/4e7a8190-3db0-490c-8078-1b78045a980a" />

On the PickScore task, the reward peaks at around 23.2 at step = 240 and then steadily decreases, which is quite different from the curve in Fig. 3.

Could you give me some advice on reproducing the OCR and PickScore results in the paper?
Thanks!

**Comment by Luo-Yihong** [comment]
To clarify: for the reward curve comparison, both our DGPO and Flow-GRPO used a very small KL loss, in order to conduct a head-to-head comparison of their ability to improve reward. However, for the main table, we used a larger KL loss and beta_dpo to better avoid reward hacking, which is consistent with Flow-GRPO. 

Additionally, the currently public repo adopts the suggestions from the reviewers during the review process — we have introduced PPO-style clip. However, we have not extensively studied this component at the moment, and it may provide stronger stability while decreasing in-domain performance.

**Comment by Solitude-lyh** [comment]
Got it! Thanks for your detailed clarification.
